File: javascript/leetcode/169. majority element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
  const majMap = new Map();
  let mode;

  nums.forEach(val => {
    if (mode === undefined) {
      mode = val;
    }
    if (majMap.has(val)) {
      const newVal = majMap.get(val) + 1;
      majMap.set(val, newVal);

      if (majMap.get(val) > majMap.get(mode)) mode = val;
    } else majMap.set(val, 1);
  });

  return mode;
};

File: go/leetcode/169. majority element.go
func majorityElement(nums []int) int {
    maj := make(map[int]int)
    var mode int
    
    for i := 0; i < len(nums); i++ {
        number := nums[i]
        _, ok := maj[number]
        
        if i == 0 {
            mode = number 
        }
        
        if ok {
            maj[number] += 1
            if(maj[number] > maj[mode]) {
                mode = number
            }
        } else {
            maj[number] = 1
        }
    }    
    
    return mode
}